### PR TITLE
Remove pytest.fixtures to avoid pytest warnings

### PR DESCRIPTION
**Pull request by fhieber**
Removes warnings mentioned in #542 

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] Passed code style checking (`./style-check.sh`)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit/test_inference.py [hunk 1]
```diff
@@ -29,7 +29,6 @@
 _EOS = -1
 
 
-@pytest.fixture
```
Comment (tdomhan) [inline test/unit/test_inference.py]: wait, shouldn't it be enough to remove this fixture decorator? The warning was about calling the fixture directly, not about having fixtures in general.

File: test/unit/test_translate.py [hunk 2]
```diff
@@ -27,16 +25,6 @@
             "Test file line 2\n"
 
 
-@pytest.fixture
```
Comment (tdomhan) [inline test/unit/test_translate.py]: why do we want to remove these?

Comment (fhieber) [reply test/unit/test_translate.py]: Why would it matter? The test pass as is and there are no fixtures needed. Imo, these just affected readability negatively and didn't really have a useful purpose.

Comment (tdomhan) [reply test/unit/test_translate.py]: just wondering. That's true, given the one liners, they did not add a ton of value.

**Pull request merged**